x-pack/plugins/entity_manager/server/lib/queries/index.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod';
+
+export const entitySourceSchema = z.object({
+  type: z.string(),
+  timestamp_field: z.optional(z.string()).default('@timestamp'),
+  index_patterns: z.array(z.string()),
+  identity_fields: z.array(z.string()),
+  metadata_fields: z.array(z.string()),
+  filters: z.array(z.string()),
+});
+
+export type EntitySource = z.infer<typeof entitySourceSchema>;
+
+const sourceCommand = ({ source }: { source: EntitySource }) => {
+  let query = `FROM ${source.index_patterns}`;
+
+  const esMetadataFields = source.metadata_fields.filter((field) =>
+    ['_index', '_id'].includes(field)
+  );
+  if (esMetadataFields.length) {
+    query += ` METADATA ${esMetadataFields.join(',')}`;
+  }
+
+  return query;
+};
+
+const filterCommands = ({
+  source,
+  start,
+  end,
+}: {
+  source: EntitySource;
+  start: string;
+  end: string;
+}) => {
+  const commands = [
+    `WHERE ${source.timestamp_field} >= "${start}"`,
+    `WHERE ${source.timestamp_field} <= "${end}"`,
+  ];
+
+  source.identity_fields.forEach((field) => {
+    commands.push(`WHERE ${field} IS NOT NULL`);
+  });
+
+  source.filters.forEach((filter) => {
+    commands.push(`WHERE ${filter}`);
+  });
```

Comment:
We might want to push (at least) these filters down to Lucene using ES DSL instead, since we can't really control what is being passed in and thus cannot guarantee the query string will be well formed.

Reply:
Not sure I get this, do you have an example ?

Reply:
`filter` here could be any string, so we're basically just doing string concatenation which doesn't feel great. 
Mainly my point is that it would be great if we could use an object or something instead to create these pieces so that it would be safer codewise.

But, if we're doing that, we may want to just pass that directly down to Lucence as ES DSL `bool` query instead. https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-rest.html#esql-rest-filtering

Reply:
I see didn't know we could have dsl filters in addition to esql query. There's no validation anywhere atm, I'll create a follow up and include this suggestion